### ISSUE TITLE
Nextcloud community  HTTP-Transmission Error

### ISSUE THREAD

**Issue by dacoex** [nextcloud/desktop#908]
### Expected behaviour
Normal synching

### Actual behaviour
Client throws an error:
"HTTP-Transmission Error"

See also:
https://help.nextcloud.com/t/http-transmission-error/41620

### Steps to reproduce
1. Install client 2.5
2. Synch
3.

### Client configuration
Client version: 2.5.0git

Operating system: 4.15.0-39-generic / Linux Mint 19 Tara / Ubuntu bionic

OS language: german

Qt version used by client package (Linux only, see also Settings dialog):
?

Client package (From Nextcloud or distro) (Linux only):

Installation path of client:

### Server configuration

Operating system: ?

Web server: ?

Database: ?

PHP version: 5

Nextcloud version: 14.02

Storage backend (external storage):

### Logs

Please use Gist (https://gist.github.com/) or a similar code paster for longer
logs.

```Template for output < 10 lines```

1. Client logfile: Output of `nextcloud --logwindow` or `nextcloud --logfile log.txt`
(On Windows using `cmd.exe`, you might need to first `cd` into the Nextcloud directory)
(See also https://docs.nextcloud.com/desktop/2.3/troubleshooting.html#log-files)

2. Web server error log:

3. Server logfile: nextcloud log (data/nextcloud.log):

**Comment by Guglio95** [comment]
+1

**Comment by rodrigoscoelho** [comment]
Same problem here. Windows Cliente. Tested on 2.5.1 with Nextcloud 15, same problem...

**Comment by splashote** [comment]
Same here I guess.

https://github.com/nextcloud/desktop/issues/819

**Comment by harcesz** [comment]
same problem, tested o 2-3 stations, good thing, we havent upgraded all our clients yet... 

a simple workaround for ubuntu/linux is to use the 3.3 appimage; https://download.nextcloud.com/desktop/releases/Linux/Nextcloud-2.3.3-x86_64.AppImage

**Comment by im-fabian** [comment]
Same here. But it seems that nobody is on that issue.

**Comment by ruihildt** [comment]
Same issue, doesn't work with 2.5.2, but working with 2.3.x

**Comment by patrickatamaniuk** [comment]
same here, not working with 2.3.3 client on mac either

**Comment by jmcclelland** [comment]
I am getting this error when I try to sync an account with orphaned files on the server. The files are present in the file system, but not in the database.

**Comment by DBJRdev** [comment]
Hi, as this issue seems to cause trouble for several users, I want to share how we got rid of the annoying problem: for us the solution came simply when emptying the trash bin one-by-one. We are not sure what had caused the problem, but at least the clients are able to sync again now.

NC 16.0.1
Client: 2.5.2git

**Comment by Pato98** [comment]
Same here. 
NC 16.0.1
Client 2.5.2

**Comment by decoppetjoris** [comment]
Same issue with client 2.5.2. It was working but juste after the update the file doesn't synchronise.

**Comment by brunt82** [comment]
The client scans the whole user directory on the server before starting the sync process. If there is any 
directory (mostly a share), which cannot be opened, the client stops with the described error message. 

Why the share cannot not be opened may have different reasons. Anyhow unsharing the share will help. 

Because the client does not give any advice in the error message, which folder / share makes the problem, you should start the client via command line adding the parameter "--logwindow".

There you should find similar to following:

> [OCC::LsColJob::finished 	LSCOL of QUrl("https://nextcloud.domain.de/remote.php/dav/files/user1/") FINISHED WITH STATUS "OK"
[OCC::DiscoverySingleDirectoryJob::directoryListingIteratedSlot 	Missing properties: "Folder123" 2 0 1561618112 "DNVS" "" "-0000001ocl3ir5nxzae"
[csync_ftw 	opendir failed for  - errno 10011
[OCC::SyncEngine::handleSyncError 	ERROR during  csync_update :  "Es hat sich ein HTTP-Übertragungsfehler ereignet. Bei der Server-Dateierkennungsantwort fehlen Daten."

"Folder123" is the share, which makes the problem... but maybe there is not only one.

**Comment by voodoomatz** [comment]
Same here.
NC 16.0.3
Client 2.5.1final (build 20181204) MacOS

**Comment by kyhuynhduc** [comment]
![image](https://user-images.githubusercontent.com/36893030/62116282-1e2c0600-b2e4-11e9-864c-e168b93242eb.png)
how can we fix?

**Comment by Highview-Lucas** [comment]
I just had this issue - the problem was (User A) had created a folder, and shared it with (User B) who then sync'd it to their devices (one windows, one linux) 

User A then decided they no longer needed the folder that had been shared, but both clients were not aware that the file not only had been unshared, but completely did not exist anymore. 

Luckily for me, the NC recycle bin works pretty well. Restoring the deleted share allowed (User B)'s clients to sync again. 

While that error was happening, no sync was happening at all.

**Comment by rinchen** [comment]
#metoo  :-)   This thread, and the linked one to 819, helped.  By referencing the log in the UI I was able to see what folder was erroring. I then had the user go into trash and restore that folder. The desktop client error went away and syncing resumed.

**Comment by gvansanden** [comment]
Very reproducable also with NC 16 and Sync client 2.5.3 on all platforms.  
Share a folder from user A, have B sync it.  Delete folders in it and broken

**Comment by maiwann** [comment]
Same issue here, even when selecting no folder to sync.

It's on a fresh account, synced between 2 computers.

Nextcloud 15, client 2.5.3.

**Comment by gvansanden** [comment]
If I may suggest, would it not be possible to make a manual locking mechanism in the interim?  I'm sure it's better to have people manually check out (lock) documents before editing and having that lock activate in all clients that are currently connected.  It would not solve the issue of disconnected users (nothing can), it would not eliminate conflicts either, but it could drastically reduce their occurance.

I also disagree that home users don't require locking.  I use Nextcloud for personal data together with my wife and we did have conflicts because I'm often updating common files on the road.

**Comment by pdcemulator** [comment]
+1

**Comment by pokapow** [comment]
> I removed this share in database table oc_shares and now it is working. Is there any way to check integrity from shares in database and files/folders in filesystem?

Same problem here. Can you explain how do you remove this share folder in database in NC 16 ?

**Comment by gvansanden** [comment]
@brunt82 The client should handle it better, but they are indeed server side issues.  In my case with Groupfolders (which break sync in many ways)

**Comment by brunt82** [comment]
> 
> 
> > There was a broken share. Not visible in Web UI, but in Windows client. I removed this share in database table oc_shares and now it is working. Is there any way to check integrity from shares in database and files/folders in filesystem?
> 
> Indeed: I had to do the same: It seems that the web client of NC 15 doesn't display folders anymore on which the user has no access permission (because e.g. the initial domain user does not exist anymore).
> In NC 13 this folders could be unshared by the user via the web client. Now I have to delete the corresponding row in oc_shares.

It seems that the shares are still visible in the section "Shared with me". There it is possible to unshare the faulty links. In our cases it occurs if a LDAP user has been removed and therefore Nextcloud does not show his shared data anymore. Unsharing of files / folders shared by a question mark resolves the issue: ![grafik](https://user-images.githubusercontent.com/26540488/70623097-9f45c300-1c1d-11ea-96c2-710d9d305595.png)

**Comment by sisyphosloughs** [comment]
Same Problem here
NC 17.0.2.1
Version 2.6.2@Manjaro
Other Clients work: Android, MacOS, Windows (2.6.1)

**Comment by mburnicki** [comment]
This is a really nasty bug which persists for more than a year without a fix. :-(
I've recently also stumbled across this after the client was upgraded from 2.3.x to a newer version 2.5.x or 2.6.x.
My server was initially an old Owncloud instance which was migrated to NC and updated to the current version. The only solution I found was to manually fiddle with the NC data base and remove some entries referring to dead links or directories that were mentioned in the client's log window which is created if the client is started with the associated parameter (see an earlier message in this thread).

Not even the command "occ files:scan --all" run on a current NC instance deletes these dead entry from the data base.

From a project like Nextcloud I'd expect that

- The client doesn't stop syncing just because there is a tiny error. It should emit a comprehensive error message (i.e., *which* file caused the problem), and continue synchronizing the remaining files.

- The file scan on the server should not only look for new files to be added to the data base, it should also remove dead entries from the data base.

Just my opinion.

**Comment by dacoex** [comment]
updating Nextcloud & client solved it for me.

**Comment by ritsute** [comment]
How i solved the issue.

Run nextcloud Sync client using CMD: 
`C:\Program Files\Nextcloud\nextcloud.exe" --logwindow`

Use the search to search for the following string: 
`[OCC::SyncEngine::handleSyncError 	ERROR during  csync_update :  "A HTTP transmission error happened. The server file discovery reply is missing data."`

Check "Missing properties" for problem share folder.

`[OCC::DiscoverySingleDirectoryJob::directoryListingIteratedSlot 	Missing properties: "3. Billing for Agencies (10th monthly)" 2 0 1578564564 "DNVS" "" "-0000001ocyqbwbgrr3w"
[csync_ftw 	opendir failed for  - errno 10011
[OCC::SyncEngine::handleSyncError 	ERROR during  csync_update :  "A HTTP transmission error happened. The server file discovery reply is missing data."`

3. Billing for Agencies (10th monthly) is our problem shared folder.

Login to mysql in your server hosting Nextcloud.
`mysql -u <user> -p <database>`

Replace "/3. Billing for Agencies (10th monthly)" with your own problem folder.
Take note of the /. Run the following command below:
`select * from oc_share where file_target = '/3. Billing for Agencies (10th monthly)';
`

Take note of the ID.
ie. 
`| 79 |          1 | Support    | NULL     | test  | test      |   NULL | folder    | 106654      | NULL        |      106654 | /3. Billing for Agencies (10th monthly) |          31 | 1571635407 |        0 | NULL       | NULL  |         0 | NULL       |                0 | NULL |             0 | NULL  |
`

Run the following to delete that offending row.
`delete from oc_share where id = 79;`

Done

**Comment by sisyphosloughs** [comment]
@ritsute : Thank you so much! This helped me!
In my case I found a share from another user, which I removed. I did this on the nextcloud server. After that the share disappeared.

From my point of view there are two things which needs to be fixed:

1. The share removing code needs be checked. Maybe we need an occ comman, which checks, repairs or removes shares, which are in a difficult status.
2. The windows/linux/macos sync client should treat removed links, like the web server: If they are not visible, the client should also ignore it.

I start to be sad about nextcloud. It seems to me, the software is getting more and more complex, and no one is interested in taking care for elementary functions.

**Comment by pwaring** [comment]
Is there a way to diagnose which folder is causing the problem without running `--logwindow`? I've had a report from a user with this error but I don't have access to their machine to run the client in debugging/logging mode. I do have root on the server and therefore access to the error logs, database etc.

**Comment by gerroon** [comment]
I am Linux and using the appimage app. How should I go about this? This is really nasty because I cant sync anymore :(

**Comment by ruihildt** [comment]
> I am Linux and using the appimage app. How should I go about this? This is really nasty because I cant sync anymore :(

I think you just need to wait for the next version, which has a fix for this. you're quite lucky, other have been waiting for years. :D

**Comment by gvansanden** [comment]
@gerroon if this is the same issue I had, you can find the broken entries with this query:
select count(*) from oc_mounts where root_id in (select fileid from oc_filecache where path like '%__groupfolders/trash%'

A debug of your client will also tell.  Fixing them manually makes sync work again.  I had this query in Zabbix.

**Comment by pwaring** [comment]
I think my user may be having a different issue, as the solution suggested by @gvansanden didn't return any results in MySQL.

**Comment by gvansanden** [comment]
@pwaring That is indeed not the only cause of missing files, just when they are groupfolders....
There are other cases where the client still tries to get files that are no longer there, I don't have a query to detect those though.  But it should be possible to get it from the database.

**Comment by Lange23** [comment]
For me I had the error because of nginx, also see my entry here:

https://help.nextcloud.com/t/http-transmission-error/41620/21?u=j4mb4l4j4

**Comment by SchmelAtWork** [comment]
Have the same issue with latest 19 rc3 server and daily build client. But in my case the reason of this error was external storage connected via SMB. After I removed these storages I do about 20 runs of client without errors.
Server published via HAproxy (http -> HAproxy -> https). if it can help.

```
[OCC::DiscoverySingleDirectoryJob::directoryListingIteratedSlot 	Missing properties: "J$" 2 0 1590162768 "M" "" "-0000001oc2zaihzqu9l"
[OCC::DiscoverySingleDirectoryJob::directoryListingIteratedSlot 	Missing properties: "IT$" 2 0 1590162768 "DNVM" "" "-0000001oc2zaihzqu9l"
[OCC::DiscoverySingleDirectoryJob::directoryListingIteratedSlot 	Missing properties: "A$" 2 0 1590162768 "DNVM" "" "-0000001oc2zaihzqu9l"
[csync_ftw 	opendir failed for  - errno 10011
[OCC::SyncEngine::handleSyncError 	ERROR during  csync_update :  "A HTTP transmission error happened. The server file discovery reply is missing data."
[OCC::User::slotAddError 	Item  "B:\\!_Cloud\\Nexcloud"  retrieved resulted in  "A HTTP transmission error happened. The server file discovery reply is missing data."
[OCC::ActivityListModel::addErrorToActivityList 	Error successfully added to the notification list:  "A HTTP transmission error happened. The server file discovery reply is missing data."
[OCC::SyncJournalDb::close 	Closing DB "C:/Users/user/AppData/Roaming/Nextcloud/._sync_efb64f3095d1.db"
[OCC::SyncEngine::finalize 	CSync run took  6898 ms
[OCC::Folder::slotSyncFinished 	Client version 2.7.0daily-Win64 (build 20200522)  Qt 5.12.5  SSL  OpenSSL 1.1.1d  10 Sep 2019
[OCC::Folder::slotSyncFinished 	SyncEngine finished with ERROR
[OCC::Folder::showSyncResultPopup 	Folder sync result:  2
[OCC::Folder::slotSyncFinished 	the last 3 syncs failed
[OCC::ownCloudGui::slotSyncStateChange 	Sync state changed for folder  "https://f.mydomain.ru/remote.php/dav/files/1A4BEC94-00F4-40ED-80F7-0CAFDA140D11/" :  "Error"
[OCC::FolderMan::slotFolderSyncFinished 	<========== Sync finished for folder [B:\!_Cloud\Nexcloud] of account [user@mydomain.ru@f.jgl.ru] with remote [https://f.mydomain.ru/remote.php/dav/files/1A4BEC94-00F4-40ED-80F7-0CAFDA140D11/]
```

**Comment by cknoll** [comment]
I had the same or a related issue. However I have no access to the server. The problem was that I had a share with a user who was removed from the server. Then my client (version 2.5.1 on debian 10) refused to synch. I solved the problem via: `nextcloud --logwindow`, searched for `Missing properties` as explained by @ritsute above and that way identified the problematic shared folder. Then I unshared it via the web interface, which worked all the time. After the client synced again.
